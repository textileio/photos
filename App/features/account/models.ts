import { ICafeSession } from '@textile/react-native-sdk'

export interface CafeSessionData {
  readonly session?: ICafeSession
  readonly processing: boolean
  readonly error?: string
}

export interface CafeSessionsData {
  readonly [peerId: string]: CafeSessionData | undefined
}
