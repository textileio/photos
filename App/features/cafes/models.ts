import { ICafeSession } from '@textile/react-native-sdk'

export interface Cafe {
  readonly peerId: string
  readonly token?: string // this is optional because we may not have tokens for previously registered custom cafes
  readonly error?: string
  readonly state: 'registering' | 'registered' | 'deregistering'
}

export interface Cafes {
  readonly [peerId: string]: Cafe
}

export interface CafeSessionData {
  readonly session: ICafeSession
  readonly processing: boolean
  readonly error?: string
}

export interface CafeSessionsData {
  readonly [peerId: string]: CafeSessionData
}
