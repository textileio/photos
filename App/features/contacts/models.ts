import { pb } from '@textile/react-native-sdk'
import Contacts from 'react-native-contacts'

export interface AddingContact {
  readonly error?: string
}

export interface AddingContacts {
  readonly [key: string]: AddingContact
}

export interface TextileSearchResult {
  readonly key: string
  readonly type: 'textile'
  readonly data: {
    contact: pb.IContact,
    isContact: boolean,
    adding: boolean
  }
}

export interface AddressBookSearchResult {
  readonly key: string
  readonly type: 'addressBook'
  readonly data: Contacts.Contact
}

export interface ErrorSearchResult {
  readonly key: string
  readonly type: 'error'
  readonly data: string
}

export interface LoadingSearchResult {
  readonly key: string
  readonly type: 'loading'
}

export interface EmptySearchResult {
  readonly key: string
  readonly type: 'empty'
}

export type SearchResult = TextileSearchResult | AddressBookSearchResult | ErrorSearchResult | LoadingSearchResult | EmptySearchResult

export interface SearchResultsSection {
  readonly key: string
  readonly title: string
  readonly data: SearchResult[]
}
