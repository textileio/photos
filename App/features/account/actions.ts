import { createAction, createAsyncAction } from 'typesafe-actions'
import { IContact } from '@textile/react-native-sdk'

import { SharedImage } from '../group/add-photo/models'

export const chooseProfilePhoto = createAsyncAction(
  'account/CHOOSE_PROFILE_PHOTO_REQUEST',
  'account/CHOOSE_PROFILE_PHOTO_SUCCESS',
  'account/CHOOSE_PROFILE_PHOTO_ERROR'
)<void, { image: SharedImage; data: string }, any>()

export const cancelProfilePhotoUpdate = createAction(
  'account/CANCEL_PROFILE_PHOTO_UPDATE',
  resolve => {
    return () => resolve()
  }
)

export const refreshProfileRequest = createAction(
  'account/REFRESH_PROFILE_REQUEST'
)

export const refreshProfileSuccess = createAction(
  'account/REFRESH_PROFILE_SUCCESS',
  resolve => {
    return (profile: IContact) => resolve({ profile })
  }
)

export const profileError = createAction('account/PROFILE_ERROR', resolve => {
  return (error: any) => resolve({ error })
})

export const refreshPeerIdRequest = createAction(
  'account/REFRESH_PEER_ID_REQUEST'
)

export const refreshPeerIdSuccess = createAction(
  'account/REFRESH_PEER_ID_SUCCESS',
  resolve => {
    return (peerId: string) => resolve({ peerId })
  }
)

export const refreshPeerIdError = createAction(
  'account/REFRESH_PEER_ID_ERROR',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const refreshAddressRequest = createAction(
  'account/REFRESH_ADDRESS_REQUEST'
)

export const refreshAddressSuccess = createAction(
  'account/REFRESH_ADDRESS_SUCCESS',
  resolve => {
    return (address: string) => resolve({ address })
  }
)

export const refreshAddressError = createAction(
  'account/REFRESH_ADDRESS_ERROR',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const setUsernameRequest = createAction(
  'account/SET_USERNAME_REQUEST',
  resolve => {
    return (username: string) => resolve({ username })
  }
)

export const setAvatar = createAsyncAction(
  'account/SET_AVATAR_REQUEST',
  'account/SET_AVATAR_SUCCESS',
  'account/SET_AVATAR_FAILURE'
)<SharedImage, void, any>()

export const setRecoveryPhrase = createAction(
  'account/SET_RECOVERY_PHRASE',
  resolve => {
    return (recoveryPhrase: string) => resolve({ recoveryPhrase })
  }
)
