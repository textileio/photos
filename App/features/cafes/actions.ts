import { createAsyncAction } from 'typesafe-actions'

// Registering with a new cafe requires the cafe URL and string
// With the new SDK, the URL needs to be changed to the peer id
export const registerCafe = createAsyncAction(
  'cafes/REGISTER_CAFE_REQUEST',
  'cafes/REGISTER_CAFE_SUCCESS',
  'cafes/REGISTER_CAFE_ERROR'
)<
  {
    peerId: string
    token: string
    success?: () => void
  },
  string,
  {
    peerId: string
    error: any
  }
>()

// Deregistering from a cafe requires the cafe's peer id
export const deregisterCafe = createAsyncAction(
  'cafes/DEREGISTER_CAFE_REQUEST',
  'cafes/DEREGISTER_CAFE_SUCCESS',
  'cafes/DEREGISTER_CAFE_ERROR'
)<
  {
    id: string
    success?: () => void
  },
  string,
  {
    id: string
    error: any
  }
>()
