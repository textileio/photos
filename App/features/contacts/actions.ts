import { createAction } from 'typesafe-actions'
import { pb } from '@textile/react-native-sdk'
import Contacts from 'react-native-contacts'

export const getContactsRequest = createAction('GET_CONTACTS_REQUEST')

export const getContactsSuccess = createAction('GET_CONTACT_SUCCESS', (resolve) => {
  return (contacts: ReadonlyArray<pb.IContact>) => resolve({ contacts })
})

export const searchRequest = createAction('@contacts/SEARCH_REQUEST', (resolve) => {
  return (searchString: string) => resolve({ searchString })
})

export const searchStarted = createAction('@contacts/SEARCH_STARTED')

export const searchResultTextile = createAction('@contacts/SEARCH_RESULT_TEXTILE', (resolve) => {
  return (result: pb.IContact) => resolve({ result })
})

export const textileSearchComplete = createAction('@contacts/TEXTILE_SEARCH_COMPLETE')

export const searchResultsAddressBook = createAction('@contacts/SEARCH_RESULTS_ADDRESS_BOOK', (resolve) => {
  return (results: ReadonlyArray<Contacts.Contact>) => resolve({ results })
})

export const searchErrorTextile = createAction('@contacts/SEARCH_ERROR_TEXTILE', (resolve) => {
  return (error: any) => resolve({ error })
})

export const searchErrorAddressBook = createAction('@contacts/SEARCH_ERROR_ADDRESS_BOOK', (resolve) => {
  return (error: any) => resolve({ error })
})

export const clearSearch = createAction('@contacts/CLEAR_SEARCH')

export const addContactRequest = createAction('@contacts/ADD_CONTACT_REQUEST', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const addContactSuccess = createAction('@contacts/ADD_CONTACT_SUCCESS', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const addContactError = createAction('@contacts/ADD_CONTACT_ERROR', (resolve) => {
  return (contact: pb.IContact, error: any) => resolve({ contact, error })
})

export const clearAddContact = createAction('@contacts/CLEAR_ADD_CONTACT_ERROR', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const authorInviteRequest = createAction('@contacts/AUTHOR_INVITE_REQUEST', (resolve) => {
  return (contact: Contacts.Contact) => resolve({ contact })
})
