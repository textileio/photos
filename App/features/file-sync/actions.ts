import { createAction, createAsyncAction } from 'typesafe-actions'

export const begin = createAction('file-sync/BEGIN', resolve => {
  return (group: string, id: string) => resolve({ group, id })
})

export const uploadProgress = createAction(
  'file-sync/UPLOAD_PROGRESS',
  resolve => {
    return (id: string, progress: number) => resolve({ id, progress })
  }
)

export const uploadComplete = createAction(
  'file-sync/UPLOAD_COMPLETE',
  resolve => {
    return (id: string, responseCode: string, responseBody: string) =>
      resolve({ id, responseCode, responseBody })
  }
)

export const error = createAction('file-sync/ERROR', resolve => {
  return (id: string, error: string) => resolve({ id, error })
})
