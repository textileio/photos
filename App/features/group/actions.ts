import { feedActions as feed } from './feed'
import { addMessageActions as addMessage } from './add-message'
import { addPhotoActions as addPhoto } from './add-photo'

export { feed, addMessage, addPhoto }
