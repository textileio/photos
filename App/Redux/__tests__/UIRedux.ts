import actions, { reducer } from '../UIRedux'
import { SharedImage } from '../../features/group/add-photo/models'
import { pb } from '@textile/react-native-sdk'

const initialState = reducer(undefined, {} as any)
const threadId: string = 'threadId' as any
const comment = 'comment'
const sharedImage: SharedImage = {
  isAvatar: false,
  origURL: 'origURL',
  uri: 'uri',
  path: '/here',
  canDelete: false
}

const sharedPhoto: pb.IFiles = {
  username: 'user',
  avatar: 'avatar',
  caption: '',
  author: 'a',
  block: 'b',
  comments: [],
  date: { seconds: 10, nanos: 20 },
  files: [],
  likes: [],
  target: 'd',
  threads: []
}

describe('ui stories', () => {
  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(initialState).toMatchSnapshot()
    })
  })
  describe('sharing a photo', () => {
    it('should update sharing photo via SharedPhoto', () => {
      const state = reducer(initialState, actions.updateSharingPhotoImage(sharedImage))
      expect(state.sharingPhoto!.image).toMatchObject(sharedImage)
    })
    it('should update sharing photo via photo hash', () => {
      const state = reducer(initialState, actions.updateSharingPhotoImage(sharedPhoto))
      expect(state.sharingPhoto!.image).toEqual(sharedPhoto)
    })
    it('should update sharing thread', () => {
      const state = reducer(initialState, actions.updateSharingPhotoThread(threadId))
      expect(state.sharingPhoto!.threadId).toEqual(threadId)
    })
    it('should update sharing comment', () => {
      const state = reducer(initialState, actions.updateSharingPhotoComment(comment))
      expect(state.sharingPhoto!.comment).toEqual(comment)
    })
    it('should cancel sharing', () => {
      const state = reducer(initialState, actions.updateSharingPhotoImage(sharedPhoto))
      expect(state.sharingPhoto).toBeDefined()
      const state1 = reducer(state, actions.cancelSharingPhoto())
      expect(state1.sharingPhoto).toBeUndefined()
    })
    it('sharePhotoRequest should clear sharing state', () => {
      const state0 = reducer(initialState, actions.updateSharingPhotoImage(sharedPhoto))
      expect(state0.sharingPhoto).toBeDefined()
      const state1 = reducer(initialState, actions.sharePhotoRequest('imageId', 'threadId'))
      expect(state1.sharingPhoto).toBeUndefined()
    })
    it('imageSharingError should not update state', () => {
      const state = reducer(initialState, actions.imageSharingError(new Error()))
      expect(state).toEqual(initialState)
    })
  })
})
