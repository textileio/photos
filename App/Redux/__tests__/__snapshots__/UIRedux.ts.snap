// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui stories initial state should match snapshot 1`] = `
Object {
  "likingPhotos": Object {},
}
`;
